core/ratarmountcore/compressions.py

```diff
@@ -107,6 +114,14 @@
         lambda x: x.read(4) == (0xFD2FB528).to_bytes(4, 'little'),
         lambda x: indexed_zstd.IndexedZstdFile(x.fileno()),
     ),
+    '7z': CompressionInfo(
+        ['7z', '7zip'],
```

Comment:
This was bad design on my part but you should change the filter inside `SQLiteIndexedTar._detectCompression`. Instead of `if compressionId in ['rar', 'zip']:` it might be better to use a white list with: `if compressionId not in ['bz2', 'gz', 'xz', 'zst']:`. Without this there might be problems when trying to open a .7z file with SQLiteIndexedTar.